Formula/w/whisper-cpp.rb

```diff
@@ -0,0 +1,23 @@
+class WhisperCpp < Formula
+  desc "Port of OpenAI's Whisper model in C/C++"
+  homepage "https://github.com/ggerganov/whisper.cpp"
+  url "https://github.com/ggerganov/whisper.cpp/archive/refs/tags/v1.4.0.tar.gz"
+  sha256 "b2e34e65777033584fa6769a366cdb0228bc5c7da81e58a5e8dc0ce94d0fb54e"
+  license "MIT"
+  head "https://github.com/ggerganov/whisper.cpp.git", branch: "master"
+
+  depends_on "wget" => :test
+
+  def install
+    system "make"
+    bin.install "main" => "whisper-cpp"
+  end
+
+  test do
+    system "wget", "https://github.com/ggerganov/whisper.cpp/raw/v1.4.0/models/for-tests-ggml-tiny.bin"
+    system "wget", "https://github.com/ggerganov/whisper.cpp/raw/v1.4.0/samples/jfk.wav"
```

Comment:
check out resource usage for test block in the other formulae

Reply:
we need checksum to prevent security issues

Reply:
```
  resource("samples_jfk.wav") do
    url "https://github.com/ggerganov/whisper.cpp/raw/v1.4.0/samples/jfk.wav"
    sha256 "59dfb9a4acb36fe2a2affc14bacbee2920ff435cb13cc314a08c13f66ba7860e"
  end
  resource("for-tests-ggml-tiny.bin") do
    url "https://github.com/ggerganov/whisper.cpp/raw/v1.4.0/models/for-tests-ggml-tiny.bin"
    sha256 "c486fb9f14a28b1c1dc252741a431646cc573450c900b9d9c406e10294aa01e6"
  end

  test do
    testpath.install resource("samples_jfk.wav") # For some reason gets installed as just "./samples"...
    testpath.install resource("for-tests-ggml-tiny.bin")

    system "#{bin}/whisper-cpp", "samples", "-m", "for-tests-ggml-tiny.bin"
    assert_equal 0, $CHILD_STATUS.exitstatus, "whisper-cpp failed with exit code #{$CHILD_STATUS.exitstatus}"
  end
```

For some reason the wav file gets downloaded as just "samples"... Other than that, this seems to work.